pkg/controller/controller.go

```diff
@@ -271,16 +274,15 @@ func (p *csiProvisioner) Delete(volume *v1.PersistentVolume) error {
 	if volume == nil || volume.Spec.CSI == nil {
 		return fmt.Errorf("invalid CSI PV")
 	}
-	ann, ok := volume.Annotations[provisionerIDAnn]
+	ann, ok := volume.Spec.CSI.VolumeAttributes[provisionerIDKey]
 	if !ok {
 		return fmt.Errorf("identity annotation not found on PV")
 	}
 	if ann != p.identity {
-		return &controller.IgnoredError{Reason: "identity annotation on PV does not match ours"}
+		return &controller.IgnoredError{Reason: "identity attribute on PV does not match ours"}
```

Comment:
@sbezverk you are right. If external-prov restarts, the id will be different and this delete operation will fail due to mismatch id.

Reply:
@vladimirvivien I submitted the issue about this,  appreciate a discussion on this topic sometime in near future.

Reply:
Why do we care about provisioner identity in the first place?

Reply:
After some thought I think we do not the check provisioner's identity. The selection of a provisioner is done based on StorageClass' provisioner filed, as long as it matches, the provisioner inherits all volumes created by the old provisioner and should be able to delete them. I will remove Provisioner's ID check in DeleteVolume call.